x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -571,7 +571,7 @@ protected void afterFinishOrFailure() {
 
     @Override
     public boolean maybeTriggerAsyncJob(long now) {
-        boolean triggered;
+        // threadpool: trigger_engine_scheduler if triggered from the scheduler
```

Comment:
What is the alternative (if *not* triggered from the scheduler)?

Reply:
this is also called by the transform task when it starts, e.g. as a result of `_start` or if the transform re-initializes due to a task re-location (failover or upgrade).